src/helpers/buffers/mod.rs

```diff
@@ -38,3 +38,49 @@ mod waiting {
         }
     }
 }
+
+pub trait GatherWaitingMessage {
+    fn compress_numbers(&self, numbers: &[usize]) -> String {
```

Comment:
would something like [this](https://stackoverflow.com/questions/50380352/how-can-i-group-consecutive-integers-in-a-vector-in-rust) work here?

Reply:
Yeah it works.